Update README.md (#118)
mongodb/docs-kotlin: README.md

Please file issue reports or requests at the [Documentation Jira Project](https://jira.mongodb.org/browse/DOCS).

## Building API Documentation

Build the API documentation docs for any major and minor version releases.

> We use [Dokka](https://github.com/Kotlin/dokka) to generate the API documentation and host them on GitHub Pages.

Reviewer: ```suggestion
Use [Dokka](https://github.com/Kotlin/dokka) to generate the API documentation and publish it on GitHub Pages.
```

> To execute a build, you must have Java JDK 17+ installed. You can use brew to install jEnv and OpenJDK to manage your Java versions:

Reviewer: ```suggestion
To execute a build, you must have Java JDK 17 or later installed. Use the following brew commands to install jEnv and OpenJDK to manage your Java versions:
```

```sh
brew install jenv
brew install openjdk@17
```

> The output of the `openjdk@17` includes a command that starts with `sudo ln -sfn`. Execute this command and then run the following commands to add the JDK to your environment:

Reviewer: ```suggestion
When you install openjdk, the log includes a command that starts with `sudo ln -sfn`. Execute this command to create a symbolic link. Then run the following commands to add the JDK to your environment, modifying the path and version number to match the version of OpenJDK that you installed:
```

```sh
jenv add /Library/Java/JavaVirtualMachines/openjdk-17.jdk/Contents/Home/
jenv global 17.0
```

Fork and clone the [`mongo-java-driver`](https://github.com/mongodb/mongo-java-driver) repository. Navigate to the repository and run the following `gradle` command:

```sh
./gradlew dokkaHtml
```

> This should generate the docs in `./build/docs/`

Reviewer: ```suggestion
If successful, this command generates API docs in `./build/docs/`
```

Check out the upstream `gh-pages` branch:

```sh

> git checkout gh-pages

Reviewer: Issue:
I think the instructions are missing the step to create a branch off gh-pages.
Suggestion:
Add a step that mentions creating a feature branch.

```

> If the branch doesn't exist, create a directory for the target version. For example:

Reviewer: ```suggestion
If a directory for the branch that you're building API docs for doesn't exist, create one. For example:
```

```sh
mkdir -p 4.10/apidocs/
```

Move the contents of the `./build/docs/` directory into the target version directory. For example:

```sh
mv ./build/docs/* 4.10/apidocs/
```

> Stage the newly-created docs in Git and ensure that you did not stage anything but the desired pages. For example,

Reviewer: ```suggestion
Stage the newly-created docs in Git and ensure that the staging area only contains the pages you want. For example:
```

```sh
git add 4.10/apidocs
git status
```

> Commit and push your changes for review. Once the PR is merged to the upstream `gh-pages` branch, GitHub Pages will automatically deploy the API documentation.

Reviewer: Suggestion:

I think this should be something like "Commit and push your changes to your feature branch. Then, create a PR that compares changes against the original repository's gh-pages branch. After you get approval and merge the PR, your changes are automatically published to GitHub Pages."

> To build the Kotlin API documentation along with all of the other JVM documentation (Java, Scala, RS), see the instructions in the `docs-java-other` [README file](https://github.com/mongodb/docs-java-other#building-api-documentation).

Reviewer: ```suggestion
To build all the API documentation for all drivers, including Kotlin, that are based on the Java driver, see the `docs-java-other` [README file](https://github.com/mongodb/docs-java-other#building-api-documentation).
```
